spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/ehcache/CacheManagerTest.java

```diff
@@ -0,0 +1,47 @@
+package org.springframework.boot.autoconfigure.ehcache;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.boot.autoconfigure.PropertyPlaceholderAutoConfiguration;
+import org.springframework.boot.test.EnvironmentTestUtils;
+import org.springframework.cache.ehcache.EhCacheCacheManager;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Tests for {@link org.springframework.cache.CacheManager}.
+ *
+ * @author Eddú Meléndez
+ */
+public class CacheManagerTest {
+
+    private AnnotationConfigApplicationContext context;
+
+    @Before
+    public void init() {
+        context = new AnnotationConfigApplicationContext();
+    }
+
+    @After
+    public void close() {
+        if (this.context != null) {
+            this.context.close();
+        }
+    }
+
+    @Test
+    public void testCache() {
+        EnvironmentTestUtils
+            .addEnvironment(this.context, "ehcache.location:ehcache/ehcache-override.xml");
+        EnvironmentTestUtils.addEnvironment(this.context,
+            "ehcache.shared:true");
+        this.context.register(EhCacheCacheManager.class,
+            PropertyPlaceholderAutoConfiguration.class);
+        this.context.refresh();
```

Comment:
Please move that block to a `load` method that is responsible to create and refresh the  context. Check `HornetQAutoConfigurationTests` for an example.